### PR TITLE
new repo: create cmm.studio folder, add saka qmk firmware

### PR DESCRIPTION
**Pull request by CMMS-Freather**
new folder for cmm.studio line up keyboard
added saka68 keyboard qmk and via firmware support

## Description
added qmk firmware and via firmware for cmm.studio saka68 keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [] My change requires a change to the documentation.
- [] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/cmm.studio/saka68/firmware/cmm_studio_saka68_design_file.json [hunk 1]
```diff
@@ -0,0 +1,183 @@
+{
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/firmware/cmm_studio_saka68_design_file.json:1]: Please delete this file.

File: keyboards/cmm.studio/saka68/config.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x434D
+#define PRODUCT_ID      0x534B
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    CMM.Studio
+#define PRODUCT         CMM.Studio Saka68
+#define DESCRIPTION     CMM.Studio Saka68
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/config.h:27]:
```suggestion
#define PRODUCT         Saka68
```

File: keyboards/cmm.studio/saka68/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        KC_LCTL,   KC_LGUI,   KC_LALT,                       KC_SPC,                              KC_RALT,  MO(1),   KC_RCTL,        KC_LEFT, KC_DOWN, KC_RGHT
+    ),
+
+    [1] = LAYOUT(
+        KC_TRNS,  KC_F1,   KC_F2,   KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,    KC_F11, KC_F12,  KC_TRNS,            KC_TRNS,  KC_TRNS,
+        KC_TRNS,      KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,        KC_TRNS,  KC_TRNS,
+        KC_TRNS,       KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,            KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,                   KC_TRNS,
+        KC_TRNS,   KC_TRNS,   KC_TRNS,                       KC_TRNS,                              KC_TRNS,  KC_TRNS,   KC_TRNS,        KC_TRNS, KC_TRNS, KC_TRNS
+    ),
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/keymaps/via/keymap.c:34]: You need two more empty layers here.

File: keyboards/cmm.studio/saka68/keymaps/via/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+VIA_ENABLE = yes
+# generated by KBFirmware JSON to QMK Parser
+# https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/keymaps/via/rules.mk:25]:
```suggestion
VIA_ENABLE = yes
MOUSEKEY_ENABLE = yes
```

File: keyboards/cmm.studio/saka68/rules.mk [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+COMMAND_ENABLE = no        # Commands for debug and configuration
+
+VIA_ENABLE = yes
+
+SRC += keyboards/cmm.studio/saka68/saka68.c
+
+RAW_ENABLE = yes
+DYNAMIC_KEYMAP_ENABLE = yes
+# generated by KBFirmware JSON to QMK Parser
+# https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/rules.mk:23]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please also replace the `.` in the vendor folder name with a `_`.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cmm.studio/saka68/keymaps/via/keymap.c (hunk 3 above)
Comment (CMMS-Freather) [reply keyboards/cmm.studio/saka68/keymaps/via/keymap.c:34]: updated

File: keyboards/cmm.studio/saka68/firmware/cmm_studio_saka68_design_file.json (hunk 1 above)
Comment (CMMS-Freather) [reply keyboards/cmm.studio/saka68/firmware/cmm_studio_saka68_design_file.json]: updated

File: keyboards/cmm.studio/saka68/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+        KC_TRNS,   KC_TRNS,   KC_TRNS,                       KC_TRNS,                              KC_TRNS,  KC_TRNS,   KC_TRNS,        KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [3] = LAYOUT(
+        KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS,            KC_TRNS,  KC_TRNS,
+        KC_TRNS,      KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,        KC_TRNS,  KC_TRNS,
+        KC_TRNS,       KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,            KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,                   KC_TRNS,
+        KC_TRNS,   KC_TRNS,   KC_TRNS,                       KC_TRNS,                              KC_TRNS,  KC_TRNS,   KC_TRNS,        KC_TRNS, KC_TRNS, KC_TRNS
+    ),
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/keymaps/default/keymap.c:50]:
```suggestion
```

The default keymap doesn't need the extra layers.

File: keyboards/cmm.studio/saka68/readme.md [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# CMM.Studio Saka68
+
+CMM.Studio Saka68 Keyboard
+
+Keyboard Maintainer: [CMM.Studio Freather](https://github.com/CMMS-Freather)  
+Hardware Supported: PCB, Atmega32u4
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/readme.md:6]:
```suggestion
* Keyboard Maintainer: [CMM.Studio Freather](https://github.com/CMMS-Freather)
* Hardware Supported: PCB, Atmega32u4
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cmm.studio/saka68/config.h [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+ */
+#define MATRIX_ROW_PINS { D1, D0, B0, F6, F7 }
+#define MATRIX_COL_PINS { D4, D6, D7, B4, B5, B6, C6, F5, F4, F1, F0, B1, B2, B3, D2, D3, D5 }
+
+#define DIODE_DIRECTION COL2ROW
+
+#define DEBOUNCE 5
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
+
```
[the remark this answers is not in the document]
Comment (CMMS-Freather) [reply keyboards/cmm.studio/saka68/config.h]: updated

**Comment by CMMS-Freather** [comment]
thank you Ryan for the review. First PR, sorry for some stupid mistakes

File: keyboards/cmm_studio/saka68/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        KC_TRNS,   KC_TRNS,   KC_TRNS,                       KC_TRNS,                              KC_TRNS,  KC_TRNS,   KC_TRNS,        KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [3] = LAYOUT(
+        KC_TRNS,  KC_TRNS,   KC_TRNS,   KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS,            KC_TRNS,  KC_TRNS,
+        KC_TRNS,      KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,        KC_TRNS,  KC_TRNS,
+        KC_TRNS,       KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS,            KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS,    KC_TRNS, KC_TRNS,  KC_TRNS, KC_TRNS,                   KC_TRNS,
+        KC_TRNS,   KC_TRNS,   KC_TRNS,                       KC_TRNS,                              KC_TRNS,  KC_TRNS,   KC_TRNS,        KC_TRNS, KC_TRNS, KC_TRNS
+    ),
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/keymaps/default/keymap.c:50]:
```suggestion
```

File: keyboards/cmm_studio/saka68/readme.md [hunk 10]
```diff
@@ -0,0 +1,12 @@
+# CMM.Studio Saka68
+
+CMM.Studio Saka68 Keyboard
+
+Keyboard Maintainer: [CMM.Studio Freather](https://github.com/CMMS-Freather)  
+Hardware Supported: PCB, Atmega32u4
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/readme.md:6]:
```suggestion
* Keyboard Maintainer: [CMM.Studio Freather](https://github.com/CMMS-Freather)
* Hardware Supported: PCB, Atmega32u4
```

File: keyboards/cmm_studio/saka68/readme.md [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make cmm.studio/saka68:default
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/readme.md:10]:
```suggestion
    make cmm_studio/saka68:default
```

File: keyboards/cmm.studio/saka68/keymaps/default/keymap.c [hunk 12]
```diff
@@ -0,0 +1,35 @@
+/* Copyright 2020 CMM.Studio Freather
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/keymaps/default/keymap.c:1]: This file should be deleted.

File: keyboards/cmm.studio/saka68/readme.md [hunk 13]
```diff
@@ -0,0 +1,12 @@
+# CMM.Studio Saka68
```
Comment (fauxpark) [inline keyboards/cmm.studio/saka68/readme.md:1]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cmm_studio/saka68/readme.md (hunk 11 above)
Comment (CMMS-Freather) [reply keyboards/cmm_studio/saka68/readme.md:10]: updated

File: keyboards/cmm.studio/saka68/readme.md (hunk 13 above)
Comment (CMMS-Freather) [reply keyboards/cmm.studio/saka68/readme.md]: deleted the whole folder of cmm.studio. now only have cmm_studio folder

File: keyboards/cmm.studio/saka68/keymaps/default/keymap.c (hunk 12 above)
Comment (CMMS-Freather) [reply keyboards/cmm.studio/saka68/keymaps/default/keymap.c]: deleted the whole folder of cmm.studio. now only have cmm_studio folder

**Comment by CMMS-Freather** [comment]
is there anything I need to do?

**Review by CMMS-Freather — COMMENTED** [review]
this cmm.studio folder already been deleted. now only uses cmm_studio folder

File: keyboards/cmm_studio/saka68/readme.md [hunk 14]
```diff
@@ -0,0 +1,12 @@
+# CMM.Studio Saka68
+
+CMM.Studio Saka68 Keyboard
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/readme.md:3]: Do you have an image of the board you'd like to add here? We prefer an imgur link, with `l` appended to the filename to make it large thumbnail sized.

File: keyboards/cmm_studio/saka68/config.h [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+// This will automatically reset the EEPROM with defaults
+// and avoid loading invalid data from the EEPROM
+#define EEPROM_VERSION 0x08
+#define EEPROM_VERSION_ADDR 34
+
+// Dynamic keymap starts after EEPROM version
+#define DYNAMIC_KEYMAP_EEPROM_ADDR 35
+// Dynamic macro starts after dynamic keymaps (35+(4*10*6*2)) = (35+480)
+#define DYNAMIC_KEYMAP_MACRO_EEPROM_ADDR 680    // **** CHANGE THIS BASED ON MATRIX_ROWS & MATRIX_COLS ****
+#define DYNAMIC_KEYMAP_MACRO_EEPROM_SIZE 344    // **** CHANGE THIS BASED ON 1024-DYNAMIC_KEYMAP_MACRO_EEPROM_ADDR ****
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/config.h:64]: I'm not really sure what all this is for. I don't think you need to override any of it.

Comment (CMMS-Freather) [reply keyboards/cmm_studio/saka68/config.h:64]: this was used for via integration. I cannot have via recognize the custome vid and pid without this line of code. this was added under via's guide of wilba's pcb as refrence

File: keyboards/cmm_studio/saka68/readme.md (hunk 14 above)
Comment (CMMS-Freather) [reply keyboards/cmm_studio/saka68/readme.md:3]: added.

**Review by CMMS-Freather — COMMENTED** [review]
resolved for the image in readme

File: keyboards/cmm_studio/saka68/readme.md (hunk 14 above)
Comment (CMMS-Freather) [reply keyboards/cmm_studio/saka68/readme.md:3]: > Do you have an image of the board you'd like to add here? We prefer an imgur link, with `l` appended to the filename to make it large thumbnail sized.

dont know why the link is not working. is it necessary?

File: keyboards/cmm_studio/saka68/config.h [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+// EEPROM usage
+
+// TODO: refactor with new user EEPROM code (coming soon)
+#define EEPROM_MAGIC 0x451F
+#define EEPROM_MAGIC_ADDR 32
+// Bump this every time we change what we store
+// This will automatically reset the EEPROM with defaults
+// and avoid loading invalid data from the EEPROM
+#define EEPROM_VERSION 0x08
+#define EEPROM_VERSION_ADDR 34
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/config.h:58]:
```suggestion
```
These definitely are not used anywhere in the repo, so will not do anything.

File: keyboards/cmm_studio/saka68/readme.md (hunk 14 above)
Comment (fauxpark) [reply keyboards/cmm_studio/saka68/readme.md:3]: The image seems to have been removed.

File: keyboards/cmm_studio/saka68/config.h (hunk 15 above)
Comment (fauxpark) [reply keyboards/cmm_studio/saka68/config.h:64]: I'm not sure what you mean by that. I can't find this in the VIA docs.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cmm_studio/saka68/config.h (hunk 15 above)
Comment (fauxpark) [reply keyboards/cmm_studio/saka68/config.h:64]: Oh, I  guess you must mean this: https://gist.github.com/nooges/5878b2648b6daeefb5c44d4ec16bf912
This guide is a little outdated, you should not need to add these defines anymore.

File: keyboards/cmm_studio/saka68/readme.md [hunk 17]
```diff
@@ -0,0 +1,12 @@
...
+![CMM.Studio Saka68](https://imgur.com/gallery/jwmyRoq)
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/readme.md]:
```suggestion
![CMM.Studio Saka68](https://i.imgur.com/gZ9Thjel.png)

```
Needs to be a direct link.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cmm_studio/saka68/readme.md [hunk 18]
```diff
@@ -0,0 +1,12 @@
+# CMM.Studio Saka68
+
+CMM.Studio Saka68 Keyboard
+![CMM.Studio Saka68](<a href="//imgur.com/a/jwmyRoq">saka solder pcb</a></blockquote>
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/readme.md:4]:
```suggestion
![CMM.Studio Saka68](https://i.imgur.com/gZ9Thjel.png)

```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cmm_studio/saka68/readme.md [hunk 19]
```diff
@@ -0,0 +1,14 @@
+# CMM.Studio Saka68
+
+CMM.Studio Saka68 Keyboard
+
+solder version pcb view can be seen here
+(<a href="//imgur.com/a/jwmyRoq">saka solder pcb</a></blockquote>)
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/readme.md:6]:
```suggestion
![](https://i.imgur.com/gZ9Thjel.png)

```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/cmm_studio/saka68/readme.md (hunk 19 above)
Comment (CMMS-Freather) [reply keyboards/cmm_studio/saka68/readme.md]: done update. thank you

File: keyboards/cmm_studio/saka68/readme.md [hunk 20]
```diff
@@ -2,8 +2,8 @@
 
 CMM.Studio Saka68 Keyboard
 
-solder version pcb view can be seen here
-(<a href="//imgur.com/a/jwmyRoq">saka solder pcb</a></blockquote>)
+
+![CMM.Studio Saka68](https://i.imgur.com/gZ9Thjel.png)
```
Comment (fauxpark) [inline keyboards/cmm_studio/saka68/readme.md:6]:
```suggestion
![CMM.Studio Saka68](https://i.imgur.com/gZ9Thjel.png)

```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by CMMS-Freather — COMMENTED** [review]
this is complete. all the files are being changed per comment. compile succesful

**Review by CMMS-Freather — COMMENTED** [review]
changes have been made, and all looks good now. waiting to be merged

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/cmm_studio/saka68/readme.md [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+
+* Keyboard Maintainer: [CMM.Studio Freather](https://github.com/CMMS-Freather)
+* Hardware Supported: PCB, Atmega32u4
+
+Make example for this keyboard (after setting up your build environment):
+
+    make cmm_studio/saka68:default
+
```
Comment (Erovia) [inline keyboards/cmm_studio/saka68/readme.md:13]: Can you add "clear instructions on how to reset the board into bootloader mode" as per the [PR checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs)?

**Review by Erovia — CHANGES_REQUESTED** [review]

File: keyboards/cmm_studio/saka68/readme.md (hunk 21 above)
Comment (CMMS-Freather) [reply keyboards/cmm_studio/saka68/readme.md:13]: > Can you add "clear instructions on how to reset the board into bootloader mode" as per the [PR checklist](https://docs.qmk.fm/#/pr_checklist?id=keyboard-prs)?

updated

**Review by Erovia — APPROVED** [review]

**Pull request merged**